usbsdmux/sd_regs.py

```diff
@@ -0,0 +1,513 @@
+        "slice": (61, 59),
+        "name": "max. read current @VDD min",
+        "enum": VDD_MIN_ENUM,
+        "unit": "mA",
+    }
+    FIELDS["VDD_R_CURR_MAX"] = {
+        "slice": (58, 56),
+        "name": "max. read current @VDD max",
+        "enum": VDD_MAX_ENUM,
+        "unit": "mA",
+    }
+    FIELDS["VDD_W_CURR_MIN"] = {
+        "slice": (55, 53),
+        "name": "max. write current @VDD min",
+        "enum": VDD_MIN_ENUM,
+        "unit": "mA",
+    }
+    FIELDS["VDD_W_CURR_MAX"] = {
+        "slice": (52, 50),
+        "name": "max. write current @VDD max",
+        "enum": VDD_MAX_ENUM,
+        "unit": "mA",
+    }
+    FIELDS["C_SIZE_MULT"] = {"slice": (49, 47), "name": "device size multiplier", "convert": lambda v: 2 ** (v + 2)}
+
+    def get_computed(self):
+        result = super().get_computed()
+
+        device_size = self.C_SIZE * self.C_SIZE_MULT * self.READ_BL_LEN
+
+        result["device_size_bytes"] = {
+            "value": device_size,
+            "unit": "bytes",
+            "name": "device size",
+        }
+
+        return result
+
+
+class CSD_20(CSD_Common):
+    FIELDS = CSD_Common.FIELDS.copy()
+    FIELDS["C_SIZE"] = {
+        "slice": (69, 48),
+        "name": "device size",
+        "convert": lambda v: (v + 1) * 512 * 1024,
+        "unit": "bytes",
+    }
+
+
+def decode_csd(raw_hex):
+    raw = int(raw_hex, 16)
+    val = bitslice(raw, 127, 126)
+    if val == 0:
+        return CSD_10(raw_hex)
+    elif val == 1:
+        return CSD_20(raw_hex)
+    else:
+        raise ValueError(f"unknown CSD version {val}")
+
+
+class CID(RegisterDecoder):
+    @staticmethod
+    def convert_prinable(value, size):
+        result = ""
+        for i in reversed(range(size)):
+            c = chr((value >> 8 * i) & 0xFF)
+            result += c if c.isprintable() else "."
+        return result
+
+    FIELDS = {
+        "MID": {
+            "slice": (127, 120),
+            "name": "Manufacturer ID",
+            "enum": {
+                0x02: "SanDisk",
+                0x03: "SanDisk SD",
+                0x74: "Transcend",
+                0x9F: "Kingston SD",
+            },
+        },
+        "OID": {
+            "slice": (119, 104),
+            "name": "OEM/Application ID",
+            "convert": lambda v: CID.convert_prinable(v, 2),
+        },
+        "PNM": {
+            "slice": (103, 64),
+            "name": "Product name",
+            "convert": lambda v: CID.convert_prinable(v, 5),
+        },
+        "PRV": {
+            "slice": (63, 56),
+            "name": "Product revision",
+            "convert": lambda v: f"{int(v >> 4 & 15)}.{int(v & 15)}",
+        },
+        "PSN": {
+            "slice": (55, 24),
+            "name": "Product serial number",
+            "convert": int,
+        },
+        "RESERVED": {
+            "slice": (23, 20),
+        },
+        # "MDT": {
+        #    "slice": (19, 8),
+        #    "name": "Manufacturing date",
+        # },
```

Comment:
I guess this is replaced by the following two entries?
Should we just remove it instead of commenting it out?

Reply:
That's the original register from the spec, which I'd prefer to keep for future reference.